docs/source/daml/intro/8_Dependencies.rst

```diff
@@ -0,0 +1,86 @@
+
+Hashes and Identifiers
+----------------------
+
+Under the heading "DAR archive contains the following packages:" you get a similar looking list of package names, paired with only the long random string repeated. That hexadecimal string, ``887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625`` in this case, is the package hash and the primary and only identifier for a package that's guaranteed to be available and preserved. Meta information like name ("7Composing") and version ("1.0.0") help make it human readable but should not be relied upon. You may not always get DAR files from your compiler, but be loading them from a running Ledger, or get them from an artifact repository.
+
+We can see this in action. When a DAR file gets deployed to a ledger, not all meta information is preserved. 
+
+#. Note down your main package hash from running ``inspect-dar`` above
+#. Start the project using ``daml start``
+#. Open a second terminal and run ``daml ledger fetch-dar --host localhost --port 6865 --main-package-id "887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625" -o assets_ledger.dar``, making sure to replace the hash with the appropriate one.
+#. Run ``daml damlc inspect-dar assets_ledger.dar``
+
+You'll notice two things. Firstly, a lot of the dependencies have lost their names, they are now only identifiable by hash. We could of course also create a second project ``7Composing-1.0.0`` with completely different contents so even when name and version are available, package hash is the only safe identifier.
+
+That's why over the Ledger API, all types, like templates and records are identified by the triple ``(entity name, module name, package hash)``. Your client application should know the package hashes it wants to interact with. To aid that, ``inspect-dar`` also provides a machine-readable format for the information it emits: ``daml damlc inspect-dar --json assets_ledger.dar``. The ``main_package_id`` field in the resulting JSON payload is the package hash of our project.
+
+Secondly, you'll notice that all the ``*.daml``, ``*.hi`` and ``*.hie`` files are gone. This leads us to data dependencies.
+
+Dependencies and Data Dependencies
+----------------------------------
+
+Dependencies under the ``daml.yaml`` ``dependencies`` group rely on the ``*.hi``, ``*.hie``, and ``*.daml`` files. The information in these files is crucial for dependencies like the Standard Library, which provide functions, types and typeclasses.
```

Comment:
They really only rely on `.hi` files. `.daml` files are completely ignored. `.hie` files are only used for cross-SDK goto definition which only works for daml-stdlib/daml-prim so they might as well be ignored as well.

Reply:
Did not realise that only worked for daml-stdlib/daml-prim . I thought it worked for ``dependencies`` in general, but fair enough. removed it.